File: 2.1a.c
#include<stdio.h>
#include<stdlib.h>

int main(){

	double count = 0;
	int n = 0;
	double distance = 0;
	while (n<6){
		int m = 0;
		while (m<6){
			distance+=abs(n-m);
			count+=1;
			m+=1;
		}
		n+=1;
	}

    printf("Total distance is ");
	printf("%f",distance);
    double avg = distance/count;
	printf("\nAvg.= ");
	printf("%f",avg);
    /* 1.944444444444  */
}

File: 2.1a.py
# Find the average distance between two points on a --
# straight line made of discrete N (=6) points:	o---o---o---o---o---o
distance = 0
count = 0
n = 0
while n != 6:
    # (0,0)(0,1)..(6,6)
    m = 0
    while m != 6:
        distance += abs(n-m)
        count += 1
        m += 1
    n += 1
print(distance)
avg = distance/count
print('Avg.=', avg)

# 1.944444444444
